lucille-core/src/main/java/com/kmwllc/lucille/util/OpenSearchUtils.java

```diff
@@ -56,16 +58,17 @@ public static RestHighLevelClient getOpenSearchRestClient(Config config) {
 
     SSLFactory sslFactory = sslFactoryBuilder.build();
 
-    RestClientBuilder builder = RestClient.builder(new HttpHost(hostUri.getHost(), hostUri.getPort(), hostUri.getScheme()))
-        .setHttpClientConfigCallback(new RestClientBuilder.HttpClientConfigCallback() {
+    RestClient restClient = RestClient.builder(new HttpHost(hostUri.getHost(), hostUri.getPort(), hostUri.getScheme()))
+        .setHttpClientConfigCallback(new HttpClientConfigCallback() {
           @Override
           public HttpAsyncClientBuilder customizeHttpClient(HttpAsyncClientBuilder httpClientBuilder) {
             return httpClientBuilder.setDefaultCredentialsProvider(provider).setSSLContext(sslFactory.getSslContext())
                 .setSSLHostnameVerifier(sslFactory.getHostnameVerifier());
           }
-        });
+        }).build();
 
-    RestHighLevelClient client = new RestHighLevelClient(builder);
+    final OpenSearchTransport transport = new RestClientTransport(restClient, new JacksonJsonpMapper());
```

Comment:
Is there a reason why we need to use RestClientTransport here instead of the default ApacheHttpClient5Transport?
https://opensearch.org/docs/latest/clients/java/

if so, should add a code comment with brief explanation

Reply:
The following forum post suggests that the RestClientTransport will be deprecated with OpenSearch 3.0. It also has HTTP/2.0 (with OpenSearch 3.0) which is preferable. Will push implementation with ApacheClient soon. 

https://forum.opensearch.org/t/difference-between-apache-httpclient-5-transport-and-restclient-transport/14650

Reply:
Changes relevant to this have been pushed